test/cpp/test_ir.cpp

```diff
@@ -118,6 +123,42 @@ TEST(IrTest, TestSizeNode) {
   });
 }
 
+TEST(IrTest, TestSizeNodeDynamic) {
+  // t1 = torch.tensor(0).expand(10,10)
+  // t1[0][0] = 1
+  // res = t1.non_zero()
+  torch::lazy::Value scalar_value =
```

Comment:
nit: maybe a small comment that you are using updateSlice to insert some 1s into otherwise 0s tensor?

Reply:
I will update this in the following pr.